sdk/clientcore/annotation-processor/src/main/java/io/clientcore/tools/codegen/AnnotationProcessor.java

```diff
@@ -44,8 +47,11 @@
 import java.util.Set;
 import java.util.stream.Collectors;
 
+/**
+ * Annotation processor that generates client code based on annotated interfaces.
+ */
 @SupportedAnnotationTypes("io.clientcore.core.annotation.*")
-@SupportedSourceVersion(SourceVersion.RELEASE_8)
+@SupportedSourceVersion(SourceVersion.RELEASE_17)
```

Comment:
We should stick to Java 8 as we are changing the baseline back to 8.

Reply:
Will update this once we transition the baseline to Java 8, as it will need a few configurations currently missing in the parent POM to support that.